nettacker/core/queue_manager.py

```
@@ -0,0 +1,333 @@
+                    self._pending_tasks[dependency_key].append(task)
+
+            return None  # Task queued, will be executed later
+
+
+class CrossProcessThreadPool:
+    """
+    Manages a shared thread pool across multiple processes.
+    Allows processes to share work when they have idle threads.
+    """
+
+    def __init__(self, max_workers_per_process: int = None):
```

Ruff (0.13.3)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)